Add FallbackComponent props to ErrorBoundary

martin308 opened the pull request:
Adds a prop to the `ErrorBoundary` component called `FallbackComponent`. 

This can be used to provide a component that will be rendered when the `ErrorBoundary` component encounters an error.

The `FallbackComponent` will be passed the `error` and `info` parameters from the `componentDidCatch` function as props.

This PR can be pointed to `master` once #11 has been merged.

martin308 committed:
Render null on error

martin308 committed:
Render Fallback component on error

martin308 committed:
Provide props to FallbackComponent

martin308 committed:
Add to example

martin308 committed:
Unused parameter to satisfy the lint gods

jmshal commented on example/app.js:
```diff
@@ -95,8 +95,14 @@ class BadButton extends React.Component {
 
+// You can provide a FallbackComponent to the ErrorBoundary which will be rendered if an error is encountered
+// It will be passed the `error` and `info` from the `componentDidCatch` method as props
+const FallbackComponent = ({_error, info}) => (
```
`_error` should be `error` (even though it's not being used here, it may be a little confusing for anybody copying/pasting the examples).

bengourley reviewed (approved):
This is really awesome 🙏

I'm going to push a couple of commits…

1. addressing @jmshal's comment, because I agree – we can just comment out the entire argument object because none of it is being used.
2. some minor style nits that weren't enforced by the linter
3. some small test adjustments: some stricter/extra assertions and moving mocked (spied) function creation to within the scope of each test (the test "passes the props to the FallbackComponent" could have passed based on the behaviour of something external or in another test)

bengourley committed:
style(example): Different approach to lint error for unused example

bengourley committed:
style(src): Small style nits

bengourley committed:
test(src): Improve <ErrorBoundary /> tests

- Add more assertions
- Move mock fn creation inside each test to reduce likelihood of side
effects
- Add stricter check to <FallbackComponent /> props test
